File: app/frontend/app/models/snapshot.js
import Ember from 'ember';
import DS from 'ember-data';
import CoughDrop from '../app';
import i18n from '../utils/i18n';
import Utils from '../utils/misc';

CoughDrop.Snapshot = DS.Model.extend({
  user_id: DS.attr('string'),
  name: DS.attr('string'),
  start: DS.attr('string'),
  end: DS.attr('string'),
  device_id: DS.attr('string'),
  location_id: DS.attr('string')
});

export default CoughDrop.Snapshot;

File: lib/json_api/snapshot.rb
module JsonApi::Snapshot
  extend JsonApi::Json
  
  TYPE_KEY = 'snapshot'
  DEFAULT_PAGE = 50
  MAX_PAGE = 100
    
  def self.build_json(snapshot, args={})
    json = {}
    
    json['id'] = snapshot.global_id
    json['name'] = snapshot.settings['name'] || "Unnamed Snapshot"
    json['user_id'] = snapshot.related_global_id(snapshot.user_id)
    ['start', 'end', 'device_id', 'location_id'].each do |key|
      json[key] = snapshot.settings[key]
    end

    if args.key?(:permissions)
      json['permissions'] = snapshot.permissions_for(args[:permissions])
    end
    
    json
  end
end
